xfeat/cat_encoder/_basic_encoder.py

```diff
@@ -7,7 +7,7 @@
 from xfeat.utils import analyze_columns
 from xfeat.base import TransformerMixin
 from xfeat.types import XDataFrame
-
+import cudf
```

Comment:
xfeat uses `cudf` as an optional requirement. This line should be fixed for CPU users.